src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -902,11 +906,54 @@ else if (!authCheckService.isAtLeastTeachingAssistantForExercise(programmingExer
         }
     }
 
+    /**
+     * POST /programming-exercises/:exerciseId/export-instructor-respository/:repositoryTypeString : sends a test, solution or template repository as a zip file
+     * @param exerciseId The id of the programming exercise
+     * @param repositoryTypeString The type of repository to zip and send
+     * @return ResponseEntity with status
+     * @throws IOException if something during the zip process went wrong
+     */
+    @GetMapping(Endpoints.EXPORT_INSTRUCTOR_REPOSITORY)
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    @FeatureToggle(Feature.PROGRAMMING_EXERCISES)
+    public ResponseEntity<Resource> exportTestRepositoryForProgrammingExercise(@PathVariable long exerciseId, @PathVariable String repositoryTypeString) throws IOException {
```

Comment:
```suggestion
    public ResponseEntity<Resource> exportInstructorRepositoryForProgrammingExercise(@PathVariable long exerciseId, @PathVariable String repositoryTypeString) throws IOException {
```

Reply:
Also I think you can map the enum here directly, see e.g. https://www.baeldung.com/spring-enum-request-param